File: java/Algorithms/towerOfHanoi.java
package Algorithms;
public class towerOfHanoi {
    public static void solvetoh(int n, String s, String a, String e){
        if(n == 0) return;
        solvetoh( n - 1, s, e, a);
        System.out.println("Transfer disk " + n + " from " + s + " to " + e);
        solvetoh( n - 1, a, s, e);
    }
    public static void main(String[] args) {
        solvetoh(3, "A", "B", "C");
    }
}

File: c/Algorithms/towerOfHanoi.c
#include <stdio.h>

void towerOfHanoi(int n, char source, char auxiliary, char destination) {
    if (n == 0) {
        return;
    }
    towerOfHanoi(n - 1, source, destination, auxiliary);
    printf("Move disk %d from %c to %c\n", n, source, destination);
    towerOfHanoi(n - 1, auxiliary, source, destination);
}

int main() {
    towerOfHanoi(3, 'S', 'A', 'D'); // Use characters instead of strings
    return 0;
}
